### PR TITLE
refactor configs to runtimes

### PR DESCRIPTION
**Pull request by jhamman**

### CODE REVIEW

File: cmip6_downscaling/runtimes.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+from dask_kubernetes import KubeCluster, make_pod_spec
+from prefect.executors import DaskExecutor, Executor, LocalExecutor
+from prefect.run_configs import KubernetesRun, LocalRun, RunConfig
+from prefect.storage import Azure, Local, Storage
+
+# TODO: Add new config that is hybrid or local compute, but has prefect storage access (ie. what I've been using to debug. Local is now non-write permissions.)
+
+
+class BaseRuntime:
+    """Base configuration class that defines abstract methods (storage, run_config and executor) for subclasses."""
```
Comment (norlandrhagen) [inline cmip6_downscaling/runtimes.py:15]: """Base runtime class

File: cmip6_downscaling/workflows/bcsd_flow.py [hunk 2]
```diff
@@ -93,9 +91,9 @@
 
 with Flow(
     name='bcsd_config_test',
-    storage=run_config.storage,
-    run_config=run_config.run_config,
-    executor=run_config.executor,
+    storage=runtime.storage,
+    runtime=runtime.runtime,
```
Comment (norlandrhagen) [inline cmip6_downscaling/workflows/bcsd_flow.py:95]: run_config=runtime.runtime?

**Comment by norlandrhagen** [comment]
Values to move to `donfig` config.py:

 - connection_string (environment variables can be imported with donfig)
 - cache_paths (intermediate and results)
 - xpersist serializer

File: cmip6_downscaling/config.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+"""Config file used by donfig"""
+
+
+_defaults = {
+    'storage': {
+        'intermediate': {'uri': '/tmp/flow-outputs/intermediates', 'storage_options': {}},
+        'results': {'uri': '/tmp/flow-outputs/results', 'storage_options': {}},
+        'temporary': {'uri': '/tmp/cmip6/temp/', 'storage_options': {}},
```
Comment (jhamman) [inline cmip6_downscaling/config.py:8]: See #76 for a follow up that will allow more reasonable defaults.

File: cmip6_downscaling/config.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    },
+    'methods': {'bcsd': {}, 'gard': {}, 'maca': {}},
```
Comment (jhamman) [inline cmip6_downscaling/config.py:10]: @orianac - anything you think should be added here?

File: cmip6_downscaling/tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+@pytest.fixture(scope="session", autouse=True)
+def set_test_config():
+    config.set(
+        {
+            'storage.intermediate.uri': '/tmp/intermediate',
+            'storage.results.uri': '/tmp/results',
+            'storage.temporary.uri': '/tmp/temporary',
+        }
+    )
```
Comment (jhamman) [inline cmip6_downscaling/tests/conftest.py:14]: see #76 for a todo item here (this does not work as desired)

File: cmip6_downscaling/tests/test_runtimes.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    PangeoRuntime,
+    TestRuntime,
+    get_runtime,
+)
+
+
+def test_runtimes():
+
+    # cloud_runtime = CloudRuntime()
+    # assert isinstance(cloud_runtime, BaseRuntime)
```
Comment (jhamman) [inline cmip6_downscaling/tests/test_runtimes.py:13]: It would be nice to find a way to mock this runtime. Added a todo in #76

**Pull request merged**